### PR TITLE
statistics: ease the impact of stats feedback on cluster

### PR DESCRIPTION
**Pull request by eurekaka**
### What problem does this PR solve?

Fix https://github.com/pingcap/tidb/issues/17478

Problem Summary:

Statistics feedback would impose periodical read/write burden on the database. Each TiDB would dump the feedbacks collected on this instance into TiKV every 10 mins, and the stats owner TiDB instance would read the feedbacks dumped every 15 seconds. If the stats owner TiDB finds there are new feedbacks from TiKV, it would merge the feedbacks with statistics in cache, then dump all these updated statistics into TiKV. This dump operation is pretty heavy if there are bunches of feedbacks on bunches of columns/indexes, since it can be treated as a light-weight ANALYZE on a lot of tables.

### What is changed and how it works?

What's Changed:

First, reduce the amount of feedbacks generated on each TiDB by:
- decreasing default `MaxQueryFeedbackCount`;
- discarding feedbacks which have too small error rate, or too small scanned row count;
- discarding feedbacks which have overlapped ranges on the same index/column;

~Second, merge multiple insert/update/delete statements of dumping statistics into single ones, to reduce the unnecessary function call stacks and RPCs.~

How it Works:

Obviously, the first change can flow-control the statistics feedback mechanism fundamentally, but we may lose some stats accuracy incurred by feedback theoretically. ~The second change combines several small transactions into a big one, since we have controlled the amount of feedbacks using the first change, I guess this bigger transaction is supposed not to be a problem. However, the bigger transaction should have higher chances of write conflict, and it makes code harder to read, so I haven't made up my mind to keep it or not actually.~

### Related changes

- Need to cherry-pick to the release branch: if this PR is experimentally effective, we should apply it in release branches.

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below): [perf test result](https://docs.google.com/document/d/10owg8NT7DNEnaDJ-ubluhvumVF0DEFZcuV-XQwgiHpI/edit#)

Side effects

- Performance regression
    - Possible stats accurateness lose may cause potential query performance regression.

### Release note 
- Ease the impact of stats feedback on cluster

### CODE REVIEW

File: statistics/handle/update.go [hunk 1]
```diff
@@ -277,6 +269,22 @@ func (h *Handle) sweepList() {
 	h.mu.Lock()
 	h.mu.rateMap.merge(errorRateMap)
 	h.mu.Unlock()
+	h.siftFeedbacks()
+}
+
+func (h *Handle) siftFeedbacks() {
```
Comment (winoros) [inline statistics/handle/update.go:275]: what does sift mean? maybe add some comments here?

File: statistics/handle/update.go [hunk 2]
```diff
@@ -277,6 +269,22 @@ func (h *Handle) sweepList() {
...
+	sc := &stmtctx.StatementContext{TimeZone: time.UTC}
+	for k, qs := range h.feedback.Feedbacks {
+		fbs := make([]statistics.Feedback, 0, len(qs)*2)
+		for _, q := range qs {
+			fbs = append(fbs, q.Feedback...)
+		}
+		if len(fbs) == 0 {
+			continue
```
Comment (winoros) [inline statistics/handle/update.go:283]: It won't change the value of `h.feedback.Feedbacks[k]` here. Would this cause some problem?

File: statistics/handle/update.go [hunk 3]
```diff
@@ -277,6 +269,25 @@ func (h *Handle) sweepList() {
...
+		fbs := make([]statistics.Feedback, 0, len(qs)*2)
+		for _, q := range qs {
+			fbs = append(fbs, q.Feedback...)
+		}
+		if len(fbs) == 0 {
+			delete(h.feedback.Feedbacks, k)
+			continue
+		}
+		h.feedback.Feedbacks[k] = h.feedback.Feedbacks[k][:1]
+		h.feedback.Feedbacks[k][0].Feedback, _ = statistics.NonOverlappedFeedbacks(sc, fbs)
```
Comment (lzmhhh123) [inline statistics/handle/update.go:289]: Do we need to update the `Size` of `h.feedback`?

**Comment by SunRunAway** [comment]
@eurekaka Should this PR be cherry-picked to release branch?

**Comment by eurekaka** [comment]
> @eurekaka Should this PR be cherry-picked to release branch?

OK

**Pull request merged**